### ISSUE TITLE
Pin to previous scraper snapshots

### ISSUE THREAD

**Issue by mikeAdamss** [GSS-Cogs/gss-utils#198]
You can use the python `pickle` library to store an instantiated class (as a binary file) in a retrievable way.

example:
```
import pickle

from gssutils import *

scraper = Scraper(seed = "info.json")

# Store a snapshot of the populated scraper in a binary
with open('scrapeAsBytes', "wb") as f:
    pickle.dump(scraper, f)

# Load a scraper from a binary
with open('scrapeAsBytes', 'rb') as f:
    scraper = pickle.load(f)

scraper
```

I'd like to consider:
* outputting thls "scrape snapshot" binary in the /out folder when the pipelines run.
* enable adding a key to info.json, so that DM's can point to the Jenkins url for theses artifacts to effectively "pin" a pipeline to a previous scrape where needed.

Note - by using info.json to specify it, we'd be able to track pinned pipelines as part of scraper monitoring.

Basically, aiming to avoid the that scenario where the DM's are halfway through getting a dataset onto PMD then the source changes -> back to the DEs -> back to the DMS -> source changes -> back to the DEs ..... etc etc

**Comment by mikeAdamss** [comment]
@ajtucker 👋 

Not urgent, but the above makes sense to me, wouldn't mind your thoughts before I drag it onto a board.

**Comment by mikeAdamss** [comment]
closing. This functionality is getting picked up via the newer vcr.py work.
